sympy/printing/pycode.py

```diff
@@ -608,6 +608,13 @@ def _print_MatrixBase(self, expr):
             func = self._module_format('numpy.array')
         return "%s(%s)" % (func, self._print(expr.tolist()))
 
+    def _print_Identity(self, expr):
+        shape = expr.shape
+        if all([dim.is_Integer for dim in shape]):
```

Comment:
Are symbolic dim allowed for this method. If yes, then may be all may not workout, try using something like,
```
all([dim.is_Integer == True for dim in shape])
```

Reply:
The current plan is only for known integer dimensions to be supported. I'm not sure I understand your suggestion -- is there ever a case where `is_Integer` holds a non-boolean value?

Reply:
https://github.com/sympy/sympy/pull/17019#discussion_r293586558 clears everything. please resolve this conversation.